app/components/img-selector.js

```diff
@@ -0,0 +1,32 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+    thumbnail: null,
+    actions: {
+        uploadImage: function(e) {
+            var self = this;
+
+            //var canvas = this.$().find('.img-selector-canvas')[0];
+            //var ctx = canvas.getContext('2d');
+
+            var reader = new window.FileReader();
+            reader.onload = function(event){
+                /*
+                var img = new Image();
+                img.onload = function(){
+                    canvas.width = img.width;
+                    canvas.height = img.height;
+                    ctx.drawImage(img,0,0);
+                };
+                img.src = event.target.result;
+                 */
+                self.set('thumbnail', event.target.result);
+                var onSetImage = self.get('onSetImage');
```

Comment:
Rather than getting and calling the function, ember generally recommends using `sendAction(action, params)`. 

eg `this.sendAction('onSetImage', event.target.result)`

Also, I can't think of a good reason why this component *wouldn't* be passed a function (since otherwise the component is rather useless)- so the check for undefined may not be necessary.